src/rules/no_dupe_class_members.rs

```diff
@@ -0,0 +1,702 @@
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  [`bar`]() {}
+  [`bar`]() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  static get [`bar`]() {}
+  static get ['bar']() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  bar() {}
+  [`bar`]() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  get [`bar`]() {}
+  'bar'() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  static 'bar'() {}
+  static [`bar`]() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  ['constructor']() {}
+  ['constructor']() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  static [`constructor`]() {}
+  static ['constructor']() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  [123]() {}
+  [123]() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  [0x10]() {}
+  16() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  [100]() {}
+  [1e2]() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  [123.00]() {}
+  [`123`]() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  static '65'() {}
+  static [0o101]() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  [123n]() {}
+  123() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  [null]() {}
+  'null'() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  bar() {}
+  get bar() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  bar() {}
+  bar() {}
+  get bar() {}
+}
+      "#,
+      vec![(3, 2), (4, 2), (5, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  static bar() {}
+  static bar() {}
+}
+      "#,
+      vec![(3, 2), (4, 2)],
+    );
+
+    assert_lint_err_on_line_n::<NoDupeClassMembers>(
+      r#"
+class Foo {
+  set bar(value: number) {}
+  bar() {}
+}
```

Comment:
@magurotuna great tests, can you also add a test with nested class to make sure it's caught?

```ts
class Bar {
  bar() {
    class Foo {
      set bar(value: number) {}
      bar() {}
    }
  }
}
```

Reply:
Thanks for pointing out! I added more tests to see nested class.